.eslintrc.js

```diff
@@ -55,40 +55,92 @@ module.exports = {
     // Require the radix argument to parseInt to avoid unintentionally allowing hex or octal.
     radix: 'error',
 
-    // Require PascalCased class names.
-    '@typescript-eslint/class-name-casing': 'error',
+    // Naming conventions. The TypeScript rule is much more complete and configurable than ESLint's.
+    camelcase: 'off',
+    '@typescript-eslint/naming-convention': [
+      'error',
+      {
+        selector: 'default',
+        format: ['camelCase'],
+        leadingUnderscore: 'allow',
+        trailingUnderscore: 'allow',
+      },
+      {
+        selector: 'variable',
+        // Allow PascalCase for React components, UPPER_CASE for constants.
+        format: ['camelCase', 'PascalCase', 'UPPER_CASE'],
+        leadingUnderscore: 'allow',
+        trailingUnderscore: 'allow',
+      },
+      {
+        selector: 'function',
```

Comment:
According to [the docs](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/naming-convention.md), it's `any named function declaration or named function expression`. So no, I don't believe it covers arrow functions like that.

I believe arrow functions are why I ended up allowing `PascalCase` for variables, since React component names often start with capital letters. eg:

```js
const Component = () => ...
```